package/MDAnalysis/core/selection.py

```diff
@@ -678,7 +679,17 @@ def _apply(self, group):
         if not pattern:
             raise ValueError(f"{self.pattern!r} is not a valid SMARTS query")
         mol = group.convert_to("RDKIT", **self.rdkit_kwargs)
-        matches = mol.GetSubstructMatches(pattern, useChirality=True)
+        self.smarts_kwargs.setdefault("useChirality", True)
+        self.smarts_kwargs.setdefault("maxMatches", len(group) * 10)
```

Comment:
I thought we had agreed on `max(1000, n_atoms*10)` so as not to change the default in cases where you have few atoms?